src/indent.cpp

```diff
@@ -666,18 +666,44 @@ void indent_text(void)
               (pc->parent_type == CT_PP_ELSE)))
          {
             next = chunk_get_next(pc);
-            /* Hack to get the logs to look right */
-            memtype = next->type;
-            set_chunk_type(next, CT_PP_IF_INDENT);
-            indent_pse_push(frm, next);
-            set_chunk_type(next, memtype);
+            int should_indent_preproc = true;
+            chunk_t* preproc_next = chunk_get_next_nl(pc);
+            preproc_next = chunk_get_next_nblank(preproc_next);
+
+            /* Look ahead at what's on the line after the #if */
+            while (preproc_next->type != CT_NEWLINE)
+            {
+                if ((((preproc_next->type == CT_BRACE_OPEN) ||
+                    (preproc_next->type == CT_BRACE_CLOSE)) &&
+                    !cpd.settings[UO_pp_indent_brace].b) ||
+                    (preproc_next->type == CT_FUNC_DEF &&
+                    !cpd.settings[UO_pp_indent_func_def].b) ||
+                    (preproc_next->type == CT_CASE &&
+                    !cpd.settings[UO_pp_indent_case].b) ||
+                    (preproc_next->type == CT_EXTERN &&
+                    !cpd.settings[UO_pp_indent_extern].b))
+                {
+                    should_indent_preproc = false;
+                    break;
+                }
+                preproc_next = chunk_get_next(preproc_next);
+            }
 
-            /* Indent one level except if the #if is a #include guard */
-            int extra = ((pc->pp_level == 0) && ifdef_over_whole_file()) ? 0 : indent_size;
-            frm.pse[frm.pse_tos].indent     = frm.pse[frm.pse_tos - 1].indent + extra;
-            frm.pse[frm.pse_tos].indent_tab = frm.pse[frm.pse_tos - 1].indent_tab + extra;
-            frm.pse[frm.pse_tos].indent_tmp = frm.pse[frm.pse_tos].indent;
-            frm.pse[frm.pse_tos].in_preproc = false;
+            if (should_indent_preproc)
+            {
+                /* Hack to get the logs to look right */
```

Comment:
yeah... ok :) it just sounds nasty to hack the code just to make the logs look nice. but if you're just moving stuff around it's cool.